fe/fe-core/src/main/java/org/apache/doris/planner/MetaScanNode.java

```diff
@@ -0,0 +1,188 @@
+                paloRange.setTabletId(tabletId);
+
+                // random shuffle List && only collect one copy
+                List<Replica> replicas = tablet.getQueryableReplicas(visibleVersion);
+                if (replicas.isEmpty()) {
+                    LOG.error("no queryable replica found in tablet {}. visible version {}",
+                        tabletId, visibleVersion);
+                    if (LOG.isDebugEnabled()) {
+                        for (Replica replica : tablet.getReplicas()) {
+                            LOG.debug("tablet {}, replica: {}", tabletId, replica.toString());
+                        }
+                    }
+                    throw new RuntimeException("Failed to get scan range, no queryable replica found in tablet: " + tabletId);
+                }
+
+                Collections.shuffle(replicas);
+                boolean tabletIsNull = true;
+                List<String> errs = Lists.newArrayList();
+                for (Replica replica : replicas) {
+                    Backend backend = Catalog.getCurrentSystemInfo().getBackend(replica.getBackendId());
+                    if (backend == null || !backend.isAlive()) {
+                        LOG.debug("backend {} not exists or is not alive for replica {}",
+                            replica.getBackendId(), replica.getId());
+                        errs.add(replica.getId() + "'s backend " + replica.getBackendId() + " does not exist or not alive");
+                        continue;
+                    }
+                    String ip = backend.getHost();
+                    int port = backend.getBePort();
+                    TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress(ip, port));
+                    scanRangeLocation.setBackendId(replica.getBackendId());
+                    scanRangeLocations.addToLocations(scanRangeLocation);
+                    paloRange.addToHosts(new TNetworkAddress(ip, port));
+                    tabletIsNull = false;
+                }
+                if (tabletIsNull) {
+                    throw new RuntimeException(tabletId + " have no queryable replicas. err: " + Joiner.on(", ").join(errs));
+                }
+                TScanRange scanRange = new TScanRange();
+                scanRange.setPaloScanRange(paloRange);
+                scanRangeLocations.setScanRange(scanRange);
+
+                resultList.add(scanRangeLocations);
+            }
+        }
+        return resultList;
+    }
+
+    public void pullDictSlots(AggregateInfo aggInfo) {
+        Preconditions.checkState(aggInfo.getGroupingExprs().isEmpty());
+        List<FunctionCallExpr> funcExprList = aggInfo.getAggregateExprs();
+        for (FunctionCallExpr funcExpr : funcExprList) {
+            FunctionParams funcParams = funcExpr.getFnParams();
+            for (Expr expr : funcParams.exprs()) {
+                if (expr instanceof SlotRef && expr.isAnalyzed()) {
```

Comment:
when thie function called, expr should be analyzed, so we need to check `expr.isAnalyzed` ?

Reply:
removed it